src/CasCap.Common.Caching/Extensions/DI.cs

```diff
@@ -9,12 +9,20 @@ public static void AddCasCapCaching(this IServiceCollection services)
         //services.AddMemoryCache();//now added inside DistCacheService
         services.AddSingleton<IRedisCacheService, RedisCacheService>();
         services.AddSingleton<IDistCacheService, DistCacheService>();
-        services.AddSingleton<AsyncKeyedLocker<string>>();
         services.AddHostedService<LocalCacheInvalidationService>();
+        CachingOptions cachingOptions = new();
         services.AddSingleton<IConfigureOptions<CachingOptions>>(s =>
         {
             var configuration = s.GetService<IConfiguration?>();
-            return new ConfigureOptions<CachingOptions>(options => configuration?.Bind(CachingOptions.SectionKey, options));
+            return new ConfigureOptions<CachingOptions>(options => {
+                configuration?.Bind(CachingOptions.SectionKey, options);
+                cachingOptions = options;
+            });
         });
+        services.AddSingleton(new AsyncKeyedLocker<string>(o =>
```

Comment:
this should be wrapped in an extension, i.e. services.AddAsyncKeyedLocker( options...)

Reply:
It's a good suggestion for my library which I will add in my mental to-do list. Thanks. For now this is not available though.

Reply:
I tried doing this but I realized that doing so would introduce too many dependencies for what isn't really much of an advantage. Not everybody using AsyncKeyedLock will require DI.